docs/src/guidelines.md

```diff
@@ -0,0 +1,97 @@
+# [Guidelines for creating models](@id Guidelines)
+
+These are guidelines for the creation of models using NLPModels to help keeping the models uniform, and for future reference in the creation of solvers.
+
+## Bare minimum
+
+Your model should derive from `AbstractNLPModel` or some other abstract class derived from it.
+It is mandatory that it has a `meta :: NLPModelMeta` field, storing all the relevant problem information.
+The minimum information that should be set is
+
+- `nvar`: Number of variables
```

Comment:
Isn't `nvar the minimum information? The rest has default values.

Reply:
I meant the minimum information that should be set in case you're not using a simple unconstrained model. I'm gonna explain it better.